vmdb/app/models/ems_container.rb

```diff
@@ -3,15 +3,11 @@ class EmsContainer < ExtManagementSystem
     EmsKubernetes
   )
 
-  def self.types
-    subclasses.collect(&:ems_type)
-  end
-
   def self.supported_subclasses
     subclasses
   end
-
-  def self.supported_types
-    types
-  end
 end
+
+# Preload any subclasses of this class, so that they will be part of the
+#   conditions that are generated on queries against this class.
+EmsContainer::SUBCLASSES.each { |c| require_dependency Rails.root.join("app", "models", "#{c.underscore}.rb").to_s }
```

Comment:
The to_s is killing my soul.

Glad to see rails 4.1.0+ supports pathnames in require_dependency:

https://github.com/rails/rails/pull/12412
https://github.com/rails/rails/commit/2254615bff286e856de08d5f7cdf8a09e441dd6a